### DISCUSSION TITLE
How to share S3 disk storage between instances

### DISCUSSION THREAD

**Discussion by timurkh-devstark** [ClickHouse/ClickHouse#45220] [Q&A] [answered]
Hi! Sorry for my dumb question.
My case is: I have 1 clickhouse ec2 instance and using s3 bucket as clickhouse disk storage by this guide https://clickhouse.com/docs/en/guides/sre/configuring-s3-for-clickhouse-use#configure-clickhouse-to-use-the-s3-bucket-as-a-disk

I thought that if in the future my ec2 instance maybe broken/terminated/etc, so I still have all data at s3 bucket so I just need to create a new ec2 instance and use this existing s3 bucket as disk storage.

So I created another ec2 instance, configured this existing s3 bucket, connected to clickhouse server by clickhouse-client, typed "show databases" and there are no my db which I created at previous instance.

What I did wrong?

**Comment by mdonkers** [comment, marked as answer]
The s3 bucket only contains the data files. Database / tables metadata etc are all in (Zoo)keeper.

So you should either run [backups](https://clickhouse.com/docs/en/operations/backup/) (which includes the metadata), connect to the same Keeper nodes, or recreate the database+tables and connect the parts again.
